src/main/java/appeng/core/features/AEFeature.java

```diff
@@ -21,71 +21,167 @@
+	PartConversionMonitor( Constants.CATEGORY_NETWORK_BUSES ),
+
+	PortableCell( Constants.CATEGORY_PORTABLE_CELL ),
+
+	StorageCells( Constants.CATEGORY_STORAGE ),
+	MEChest( Constants.CATEGORY_STORAGE ),
+	MEDrive( Constants.CATEGORY_STORAGE ),
+	IOPort( Constants.CATEGORY_STORAGE ),
+
+	NetworkTool( Constants.CATEGORY_NETWORK_TOOL ),
+
+	DenseEnergyCells( Constants.CATEGORY_HIGHER_CAPACITY ),
+	DenseCables( Constants.CATEGORY_HIGHER_CAPACITY ),
+
+	P2PTunnelRF( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelME( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelItems( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelRedstone( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelEU( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelLiquids( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelLight( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelOpenComputers( Constants.CATEGORY_P2P_TUNNELS ),
+	P2PTunnelPressure( Constants.CATEGORY_P2P_TUNNELS ),
+
+	MassCannonBlockDamage( Constants.CATEGORY_BLOCK_FEATURES ),
+	TinyTNTBlockDamage( Constants.CATEGORY_BLOCK_FEATURES ),
+
+	Facades( Constants.CATEGORY_FACADES ),
+
+	UnsupportedDeveloperTools( Constants.CATEGORY_MISC, false ),
+	Creative( Constants.CATEGORY_MISC ),
+	GrinderLogging( Constants.CATEGORY_MISC, false ),
+	Logging( Constants.CATEGORY_MISC ),
+	IntegrationLogging( Constants.CATEGORY_MISC, false ),
+	WebsiteRecipes( Constants.CATEGORY_MISC, false ),
+	LogSecurityAudits( Constants.CATEGORY_MISC, false ),
+	Achievements( Constants.CATEGORY_MISC ),
+	UpdateLogging( Constants.CATEGORY_MISC, false ),
+	PacketLogging( Constants.CATEGORY_MISC, false ),
+	CraftingLog( Constants.CATEGORY_MISC, false ),
+	LightDetector( Constants.CATEGORY_MISC ),
+	DebugLogging( Constants.CATEGORY_MISC, false ),
+
+	enableFacadeCrafting( Constants.CATEGORY_CRAFTING ),
+	inWorldSingularity( Constants.CATEGORY_CRAFTING ),
+	inWorldFluix( Constants.CATEGORY_CRAFTING ),
+	inWorldPurification( Constants.CATEGORY_CRAFTING ),
+	InterfaceTerminal( Constants.CATEGORY_CRAFTING ),
+	enableDisassemblyCrafting( Constants.CATEGORY_CRAFTING ),
+
+	AlphaPass( Constants.CATEGORY_RENDERING ), PaintBalls( Constants.CATEGORY_TOOLS ),
+
+	MolecularAssembler( Constants.CATEGORY_CRAFTING_FEATURES ),
+	Patterns( Constants.CATEGORY_CRAFTING_FEATURES ),
+	CraftingCPU( Constants.CATEGORY_CRAFTING_FEATURES ),
+
+	ChunkLoggerTrace( Constants.CATEGORY_COMMANDS, false );
 
 	public final String category;
-	public final boolean isVisible;
 	public final boolean defaultValue;
 
-	AEFeature( String cat )
+	AEFeature( final String cat )
 	{
 		this( cat, true );
 	}
 
-	AEFeature( String cat, boolean defaultValue )
+	AEFeature( final String cat, final boolean defaultValue )
 	{
 		this.category = cat;
-		this.isVisible = !this.name().equals( "Core" );
 		this.defaultValue = defaultValue;
 	}
+
+	/**
+	 * override to set visibility
+	 *
+	 * @return default true
+	 */
+	public boolean isVisible()
+	{
+		return true;
+	}
+
+	private enum Constants
+	{
+		;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)